File: leetcode_C++/p2178.cpp
/*
@Author: mrxhar
@Date:   2023-09-06 09:11:45
@Email:  1337904145@qq.com
@Blog:   https://leetcode.cn/problems/maximum-split-of-positive-even-integers/description/?envType=daily-question&envId=2023-09-06
*/

#include <vector>
using std::vector;

class Solution {
public:
    vector<long long> maximumEvenSplit(long long finalSum) {
        vector<long long> res;
        if (finalSum % 2 != 0)
            return res;
        for (long long i = 2; i <= finalSum; i += 2) {
            res.push_back(i);
            finalSum -= i;
        }
        res.back() += finalSum;
        return res;
    }
};

int main() {
    return 0;
}

File: leetcode_py/p2178.py
from typing import List


class Solution:
    def maximumEvenSplit(self, finalSum: int) -> List[int]:
        # 贪心算法
        if finalSum % 2:
            return []
        i = 2
        res = []
        while finalSum:
            if finalSum >= 2 * i:
                res.append(i)
                finalSum -= i
            else:
                res.append(finalSum)
                finalSum = 0
            i += 2
        return res
